bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -720,8 +720,7 @@ public void asyncAddEntry(final byte[] data, final AddCallback cb,
      * @param ctx
      *            some control object
      */
-    public void asyncAddEntry(final long entryId, final byte[] data, final AddCallback cb, final Object ctx)
-            throws BKException {
+    public void asyncAddEntry(final long entryId, final byte[] data, final AddCallback cb, final Object ctx) {
```

Comment:
@eolivelli I don't remember the exact conversation I had with JV before. but it is better not to mix these two semantics together. 

I am not sure why we need an extra boolean field. You can simply override the methods and fail the requests, can't you?